shared/js/media/media_frame.js

```diff
@@ -245,7 +247,20 @@ MediaFrame.prototype.displayImage = function displayImage(blob,
     }
   }
   else {
-    noPreview();
+    // If we have valid small EXIF preview use it for css background image
+    // This will help to load large downsampled preview images without flash
+    // See bug 1188286
+    // ( TBD: preview & preview.start might not be enough.
+    // Need more robust check for small valid EXIF preview )
+    if (preview && preview.start) {
+      // Create a blob URL for the bg preview
+      self.bgPreviewURL = URL.createObjectURL(
```

Comment:
thinking about writing a comment to explain this field, it occurs to me that it might be easier to explain what this is and how it differs from this.fullImageURL and this.previewurl if you called it this.tinyPreviewURL. The size is more relevant than the fact that it is being displayed using background-image, I think.